07/hello_test.go

```diff
@@ -0,0 +1,21 @@
+package hello_test
+
+import (
+	"hello"
+	"testing"
+)
+
+var greetingTests = []string{
+	"Hi there",
+	"Hello",
+}
+
+func TestReturnGreeting(t *testing.T) {
+	for _, n := range greetingTests {
+		want := n + " yourself!"
```

Comment:
It's generally a good idea with test cases to supply the expected result as _canned data_, that is, specified as literal data, not constructed by code. (If we do construct the data with code, we shouldn't do it the same way the function under test does, just in case they're both wrong in the same way!)

So here, for example, we might write:

```go
tcs := []struct{
	input string
	want string
}{
	{
		input: "Hi there",
		want: "Hi there yourself!",
	},
	{
		input: "Hello",
		want: "Hello yourself!",
	},
}
```

It might seem trivial for this test, and it is, but it's a solid pattern that we can use for all test cases, for all tests.